lib/raxx.ex

```diff
@@ -791,13 +801,24 @@ defmodule Raxx do
   Complication with such functionality are discussed here - https://github.com/phoenixframework/phoenix/pull/1402
   Sinatra has a very complete test suite including a back implementation - https://github.com/sinatra/sinatra/blob/9bd0d40229f76ff60d81c01ad2f4b1a8e6f31e05/test/helpers_test.rb#L183
   """
+  @spec redirect(String.t(), status: status, body: body, content_type: String.t()) ::
+          Raxx.Response.t()
   def redirect(url, opts \\ []) do
     status = Keyword.get(opts, :status, :see_other)
 
-    response(status)
-    |> set_header("location", url)
-    |> set_header("content-type", "text/html")
-    |> set_body(redirect_page(url).data)
+    response = status |> response() |> set_header("location", url)
+
+    case Keyword.fetch(opts, :body) do
+      {:ok, body} ->
+        content_type = Keyword.fetch!(opts, :content_type)
```

Comment:
A response can just not have a content-type, I would be tempted to say if not set as an option don't add to request.

Docs can be updated to say content_type should be added?

Reply:
It will fail (with an exception) if `:content_type` is not given and `:body` is, since `Keyword.fetch!/2` has that behaviour.

This was also (added to docs)[https://github.com/CrowdHailer/raxx/pull/171/commits/e40a064970b033c030cb89475df3844e88a48668#diff-a06ad6f947cb23245e738a5def20d796R765] :)

Reply:
Sorry if I wasn't clear.
I'm saying not setting it shouldn't be a failure, if not set in the options list the response should just not have the content type set

Reply:
Ah, I get it ;) Will fix.

Reply:
Fixed in https://github.com/CrowdHailer/raxx/pull/171/commits/d921f6650ab0e1da14a83699e4e237a61ae9c775.

I will squash it with https://github.com/CrowdHailer/raxx/pull/171/commits/e40a064970b033c030cb89475df3844e88a48668, but I commited seperately, so you could see the changes.